LibraryApp/src/redux/login/actions.js

```diff
@@ -0,0 +1,42 @@
+import { login as loginService } from '@services/LoginService';
+import { NavigationActions } from 'react-navigation';
+import Routes from '@constants/routes';
+
+export const actions = {
+  LOGIN: 'LOGIN',
+  LOGIN_SUCCESS: 'LOGIN_SUCCESS',
+  LOGIN_FAILURE: 'LOGIN_FAILURE'
+};
+
+const privateActionCreators = {
+  loginSuccess: navigatorDispatch => dispatch => {
+    dispatch({
+      type: actions.LOGIN_SUCCESS
+    });
+    navigatorDispatch(
+      NavigationActions.navigate({
+        routeName: Routes.Library
+      })
+    );
+  },
+  loginFailure: problem => dispatch => {
+    dispatch({
+      type: actions.LOGIN_FAILURE,
+      payload: problem
+    });
+  }
+};
+
+const actionCreators = {
+  login: (email, password, navigatorDispatch) => async dispatch => {
+    dispatch({ type: actions.LOGIN });
+    const response = await loginService(email, password);
+    if (response.data.data) {
```

Comment:
I believe this can throw an error. Did you try this with failure request? I would do `response?.data?.data`

Reply:
i tried with incorrect data and the app is still working... for some reason the response always have the `data` field